nearquake/utils/__init__.py

```
@@ -231,6 +231,32 @@ def generate_date_range(start_date, end_date, interval) -> tuple:
+    :param start_date: The start date in "YYYY-MM-DD" format.
+    :param end_date: The end date in "YYYY-MM-DD" format.
+    :param interval: The interval in days between each date in the range
+
+    :return: A list or iterable of dates from start_date to end_date with the specified interval.
+    """
+
+    try:
+        datetime.strptime(start_date, "%Y-%m-%d").date()
+        datetime.strptime(end_date, "%Y-%m-%d").date()
+    except ValueError:
+        raise ValueError("Invalid date format. Use YYYY-MM-DD.")
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)